_posts/2024-10-11-tiered-cache.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title:  "Tiered caching in OpenSearch"
+authors:
+   - upasagar
+   - akjain
+   - peteral
+   - kkhatua
+   - kolchfa
+date: 2024-10-11
+categories:
+  - technical-posts
+has_science_table: true
+meta_keywords: tiered caching, disk-based caching, on-heap caching, OpenSearch caching performance, how tiered caching works
+meta_description: Explore how OpenSearch combines on-heap and disk-based caching to handle larger datasets and improve performance. Learn about the trade-offs of tiered caching, how it works, and future developments.
+---
+
+For performance-intensive applications like OpenSearch, caching is an essential optimization. Caching stores data so that future requests can be served faster, improving query response times and application performance. OpenSearch uses two main cache types: a request cache and a query cache. Both are on-heap caches, meaning their size is determined by the amount of available heap memory on a node.
+
+## On-heap caching: A good start, but is it enough?
+
+On-heap caching in OpenSearch provides a quick, simple, and efficient way to cache data locally on a node. It offers low-latency data retrieval and thereby provides significant performance gains. However, these advantages come with trade-offs, especially as the cache grows, which may lead to performance challenges.
```

Comment:
> However, these advantages come with trade-offs, especially as the cache grows, which may lead to performance challenges.

Maybe it is just me, but this line reads slightly incomplete or looks confusing

Reply:
Reworded.

Reply:
@kolchfa-aws Can we reword this to something like below? As it is more specific.

"However, these advantages come with trade-offs, especially as the cache grows in size and reaches its capacity, which may lead to performance challenges due to high evictions and misses."